templates/stack.yml

```diff
@@ -0,0 +1,204 @@
+provider:
+  name: faas
+  gateway: http://127.0.0.1:8080
+
+functions:
+  system-github-event:
+    lang: go
+    handler: ./github-event
+    image: functions/github-event:0.7.1
+    labels:
+      openfaas-cloud: "1"
+      role: openfaas-system
+    environment:
+      validate_hmac: true
+      write_debug: true
+      read_debug: true
+      validate_customers: true
+    environment_file:
+      - github.yml
+      - gateway_config.yml
+    secrets:
+      - github-webhook-secret
+      - payload-secret
+
+  github-push:
+    lang: go
+    handler: ./github-push
+    image: functions/github-push:0.7.1
+    labels:
+      openfaas-cloud: "1"
+      role: openfaas-system
+    environment:
+      validate_hmac: true
+      read_timeout: 10s
+      write_timeout: 10s
+      write_debug: true
+      read_debug: true
+    environment_file:
+      - gateway_config.yml
+      - github.yml
+    secrets:
+      - github-webhook-secret
+      - payload-secret
+
+  git-tar:
+    lang: dockerfile
+    handler: ./git-tar
+    image: functions/of-git-tar:0.9.1
+    labels:
+      openfaas-cloud: "1"
+      role: openfaas-system
+    environment:
+      read_timeout: 15m
+      write_timeout: 15m
+      write_debug: true
+      read_debug: true
+    environment_file:
+      - gateway_config.yml
+      - github.yml
+    secrets:
```

Comment:
Just curious, what ideas do you have on how to avoid templating this file?

Reply:
Is it just for the single secret that we need this?

Reply:
Yes, it's only for the secret.
I don't think OpenFaaS allows a function to use a secret without having it declared. 
One optional solution is to use secrets filters in openfaas-cloud as we're going to do here.
Another option is to always create the secret, but it's not good solution I think.

`stack.yml` will also be needed for the Swarm implementation.

Reply:
If this is an issue with the git-tar function we can only deploy the function itself i.e., after checking the oauth provider deploy just the git-tar function, by create a stack.yml file with just the git-tar function with gitlab-api-token. so there won't be a need to generate stack.yml

Reply:
Or add this function to gitlab.yml file

Reply:
Thanks for the proposal @zeerorg ! This is a good option. Would cause more refactoring once templating is moved to openfaas-cloud but for now this would work I think.